Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.Common/Providers/IGraphUtility.cs

```diff
@@ -20,83 +20,76 @@ public interface IGraphUtility
         /// <param name="teamId">The id of the team whos schedule we want to share.</param>
         /// <param name="startDateTime">The start time we want to share from.</param>
         /// <param name="endDateTime">The end time we want to share until.</param>
         /// <param name="notifyTeam">Whether we want to notify the team or not.</param>
         /// <returns>Http response message.</returns>
-        Task<HttpResponseMessage> ShareSchedule(string accessToken, string teamId, DateTime startDateTime, DateTime endDateTime, bool notifyTeam);
+        Task<HttpResponseMessage> ShareSchedule(GraphConfigurationDetails graphConfigurationDetails, string teamId, DateTime startDateTime, DateTime endDateTime, bool notifyTeam);
 
         /// <summary>
         /// Fetch user details for shifts using graph api tokens.
         /// </summary>
-        /// <param name="accessToken">Access Token.</param>
+        /// <param name="graphConfigurationDetails">The graph configuration details.</param>
         /// <returns>The shift user.</returns>
-        Task<List<ShiftTeams>> FetchShiftTeamDetailsAsync(
-            string accessToken);
+        Task<List<ShiftTeams>> FetchShiftTeamDetailsAsync(GraphConfigurationDetails graphConfigurationDetails);
+
+        /// <summary>
+        /// Sends a graph request.
+        /// </summary>
+        /// <param name="graphConfigurationDetails">The graph configuration details.</param>
+        /// <param name="httpClient">The http client.</param>
+        /// <param name="httpMethod">The http method.</param>
+        /// <param name="requestUrl">The request URI.</param>
+        /// <param name="requestString">The request string to add as the request content.</param>
+        /// <returns>A HttpResponseMessage object.</returns>
+        Task<HttpResponseMessage> SendGraphHttpRequest(GraphConfigurationDetails graphConfigurationDetails, HttpClient httpClient, HttpMethod httpMethod, string requestUrl, string requestString = null);
 
         /// <summary>
         /// Method that will get the Microsoft Graph token.
         /// </summary>
-        /// <param name="tenantId">The Tenant ID.</param>
-        /// <param name="instance">The instance.</param>
-        /// <param name="clientId">The App ID of the Configuration Web App.</param>
-        /// <param name="clientSecret">The Client Secret.</param>
-        /// <param name="userId">The user ID.</param>
+        /// <param name="graphConfigurationDetails">The graph configuration details.</param>
         /// <returns>The string that represents the Microsoft Graph token.</returns>
-        Task<string> GetAccessTokenAsync(
-            string tenantId,
-            string instance,
-            string clientId,
-            string clientSecret,
-            string userId = default(string));
+        Task<string> GetAccessTokenAsync(GraphConfigurationDetails graphConfigurationDetails);
```

Comment:
I think this should be private to the utility and if you move the addition of the auth header to within the class you can test whether GraphConfigurationDetails has a token set and if not get one all within your central method. You should be able to remove a lot of code that way.